modules/tests/shared/src/test/scala/io/circe/JsonSuite.scala

```diff
@@ -269,6 +269,30 @@ class JsonSuite extends CirceMunitSuite with FloatJsonTests {
     assert(result === expected && resultAlias === expected)
   }
 
+  test("fromStringOrNull return Null on None") {
+    assertEquals(Json.fromStringOrNull(None), Json.Null)
+  }
```

Comment:
Can you also add tests for the `Some` side here?  As it stands now we could accidentally always return null from `from*OrNull` and the tests would pass.